src/datasets/arrow_dataset.py

```diff
@@ -3464,6 +3511,11 @@ def init_buffer_and_writer():
                                 writer.write_row(example)
                             elif isinstance(example, pd.DataFrame):
                                 writer.write_row(pa.Table.from_pandas(example))
+                            elif config.POLARS_AVAILABLE:
+                                import polars as pl
+
+                                if isinstance(example, pl.DataFrame):
```

Comment:
If the example is not a `pl.DataFrame` then in your current code the line `writer.write(example)` is not called. you can group `config.POLARS_AVAILABLE` and `isinstance(example, pl.DataFrame)` in the same `if` statement to fix this instead. This way the code would continue in the `else` if the example is not a polars DataFrame.

You might have to check for `"polars" in sys.modules` in the `if`, and also add this line before `with contextlib.ExitStack() as stack`:

```python
    if config.POLARS_AVAILABLE and "polars" in sys.modules:
        import polars as pl
```